### PR TITLE
Add microprofile OpenApi annotations to JavaRxSpec (quarkus library). Add OpenID support (core)

### PR DESCRIPTION
**Pull request by miguelborges99**
Quarkus only supports Microprofile OpenApi annotations, and Swagger annotations will not be supported.
This PR intendes to generate code with Microprofile OpenApi annotations when using JAXRS-spec generator with the quarkus library option. 

Remark: In the future, these Microprofile OpenApi annotations can be extended to other libraries, if there is interest in it.

Solve issue #795

This pull request enables Microprofile OpenAPI annotations when option "useMicroProfileOpenAPIAnnotations" is set to true.
It is possible to have both Microprofile OpenAPI annotations and Swagger annotations in the same generated code.
Here is an example, using openapi-generator-maven-plugin, on how to use it:

     <plugin>
                <groupId>org.openapitools</groupId>
                <artifactId>openapi-generator-maven-plugin</artifactId>
                <version>${version.openapi.generator.maven.plugin}</version>
                <executions>
                    <execution>
                        <goals>
                            <goal>generate</goal>
                        </goals>
                        <configuration>
                            <inputSpec>${project.basedir}/../openapi.yaml</inputSpec>
                            <generatorName>jaxrs-spec</generatorName>
                            <apiPackage>com.yyy.services</apiPackage>
                            <modelPackage>com.yyy.api.model</modelPackage>
                            <invokerPackage>com.yyy.api</invokerPackage>
                            <generateSupportingFiles>false</generateSupportingFiles>
                            <generateApiTests>true</generateApiTests>
                            <generateModelTests>true</generateModelTests>
                            <generateAliasAsModel>true</generateAliasAsModel>
                            <configOptions>
                                <sourceFolder>src/main/java</sourceFolder>
                                <sourceFolder>src/gen/java</sourceFolder>
                                <java8>true</java8>
                                <library>quarkus</library>
                                <useMicroProfileOpenAPIAnnotations>true</useMicroProfileOpenAPIAnnotations>
                                <useJakartaEe>true</useJakartaEe>
                                <useSwaggerAnnotations>false</useSwaggerAnnotations>
                                <dateLibrary>java8-localdatetime</dateLibrary>
                                <serializationLibrary>jackson</serializationLibrary>
                                <interfaceOnly>true</interfaceOnly>
                                <returnResponse>true</returnResponse>
                                <useTags>true</useTags>
                            </configOptions>
                        </configuration>
                    </execution>
                </executions>
            </plugin>

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] In case you are adding a new generator, run the following additional script : 
  ```
  ./bin/utils/ensure-up-to-date.sh
  ``` 
  Commit all changed files.
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (6.3.0) (minor release - breaking changes with fallbacks), `7.0.x` (breaking changes without fallbacks)
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

@bbdouglas (2017/07) @sreeshas (2017/08) @jfiala (2017/08) @lukoyanov (2017/09) @cbornet (2017/09) @jeff9finger (2018/01) @karismann (2019/03) @Zomzog (2019/04) @lwlee2608 (2019/10)
@jmini @wing328

### CODE REVIEW

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/CodegenSecurity.java [hunk 1]
```diff
@@ -25,9 +25,10 @@
 
 public class CodegenSecurity {
     public String name;
+    public String description;
     public String type;
     public String scheme;
-    public Boolean isBasic, isOAuth, isApiKey;
+    public Boolean isBasic, isOAuth, isApiKey, isOpenId;
```
Comment (wing328) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/CodegenSecurity.java:31]: FYI @OpenAPITools/generator-core-team 

This PR also adds supports for openID security definition.

**Comment by wing328** [comment]
Tested locally and the result is good:
```
[INFO] [io.quarkus.deployment.pkg.steps.JarResultBuildStep] Building thin jar: /private/tmp/quarkus/target/openapi-jaxrs-server-1.0.0-runner.jar
[INFO] [io.quarkus.deployment.QuarkusAugmentor] Quarkus augmentation completed in 2181ms
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  6.765 s
[INFO] Finished at: 2023-05-09T17:01:58+08:00
[INFO] ------------------------------------------------------------------------
```

**Comment by miguelborges99** [comment]
Documentation was updated.

**Comment by miguelborges99** [comment]
@wing328: When this PR can be merged?

**Comment by wing328** [comment]
Looks like you made some new changes, right? https://github.com/OpenAPITools/openapi-generator/pull/15407/commits/d4eee01c259330453457a877bcc05b35469f85e2

**Comment by wing328** [comment]
Can you please create another config file based on `./bin/configs/jaxrs-spec.yaml`? e.g. `./bin/configs/jaxrs-spec-quarkus-useMicroProfileOpenAPIAnnotations.yaml` and add the option to use `useMicroProfileOpenAPIAnnotations` and the library option `quarkus`.

Then update `.github/workflows/samples-jaxrs.yaml` with the new folder of the sample to ensure it's tested in the CI when there're changes to the sample.

**Comment by miguelborges99** [comment]
Ok, no problem

**Comment by miguelborges99** [comment]
@wing328, @jimschubert: I've added the bin/configs/jaxrs-spec-quarkus-microprofile-openapi-annotations.yaml and respective sample.

**Comment by wing328** [comment]
> Note: samples-jaxrs.yaml uses Java 8, so I did not use jakarta option

FYI upcoming major version 7.0.0 will no longer support JDK8 (and default to JDK11). We'll revise the CI to test with JDK11 as part of the 7.0.0 release.

**Comment by wing328** [comment]
LGTM. Thanks for the PR.

**Pull request merged**